src/logging_manager/mod.rs

```diff
@@ -0,0 +1,340 @@
+mod time_manager;
+
+use std::fs::{create_dir, read_dir, remove_file, DirEntry, OpenOptions};
+use std::io::{Error, Write};
+use std::path::Path;
+use std::sync::mpsc;
+use std::thread;
+use log::{Metadata, Record, Level, SetLoggerError, LevelFilter};
+
+/// Struct holding sender to dedicated logging thread
+pub struct LoggingService {
+    sender: mpsc::SyncSender<LoggingCommands>,
+}
+
+// struct holding the body of a message to log
+struct MessageBody {
+    level: String,
+    content: String,
+}
+
+// commands that can be sent to the logging service
+enum LoggingCommands {
+    Kill,
+    Message(MessageBody),
+}
+
+/// Configuration struct for Logging service
+pub struct LoggingConfig {
+    /// Path from executable to directory to be used for log files
+    pub logging_directory: &'static Path,
+    /// The maximum size of the logging directory in bytes
+    pub max_dir_size: usize,
+}
+
+impl LoggingService {
+
+    /// Create a new LoggingService instance holding the sender to the dedicated logging thread.
+    ///
+    /// # Arguments
+    ///
+    /// * `options` - LoggingConfig struct containing the options for the new logging service instance.
+    ///
+    pub fn new(options: LoggingConfig) -> LoggingService {
+        let (sender, receiver) = mpsc::sync_channel(1);
+
+        // kick off logging thread
+        thread::spawn(move || loop {
+            match receiver.recv().unwrap() {
+                LoggingCommands::Message(message) => {
+                    log(message, &options)
+                        .expect("Logging service failed when receiving message.");
+                }
+                LoggingCommands::Kill => break,
+            }
+        });
+
+        LoggingService { sender }
+
+    }
+
+    /// Initiate global logger by boxing the service and sending it to the global logger.
+    ///
+    /// # Arguments
+    ///
+    /// * `max_logging_level` - LevelFilter representing the max logging level for the logging service.
+    /// Note: The order of logging levels (decreasing) is: Trace, Debug, Info, Warn, Error.
+    /// Therefore, specifying Debug as the max logging level will ignore Trace logging messages.
+    ///
+    pub fn init(self, max_logging_level: LevelFilter) -> Result<(), SetLoggerError> {
+
+        // box logger
+        let logger = Box::new(self);
+
+        // set global logger
+        log::set_boxed_logger(logger)
+            .map(|()| log::set_max_level(max_logging_level))?;
+
+        Ok(())
+    }
```

Comment:
Are we going to be setting multiple loggers?

Reply:
Well we can only set one global logger, but it would make things a bit more modular to have this code that sets the global logger in main. Right now if we wanted to use a different logger as the global logger we would basically have to copy and paste the code in this function.

Reply:
I am against using multiple loggers when we have one that is so versatile but I can move this function into main if you want

Reply:
I meant like switching between loggers for different executions of the program. Like maybe we want to have a debug configuration for the server where it doesn't write any logs to a file and instead just writes all logs to stdout. It would be easier to swap between that logger and LoggingService if we just had the log::set_boxed_logger call directly in main.

Reply:
How would we swap loggers though?

Reply:
Let me just text you lol this is weird haha